### ISSUE TITLE
cuml does not install

### ISSUE THREAD

**Issue by aph61** [MaartenGr/BERTopic#1148]
Hi,

I have problems importing (installing) cuml for my ; 
```
"class DeviceTypeError(Exception):
     23     '''An exception thrown to indicate bad device type selection'''. 
```
I've an Asus TUF Gaming GeForce RTX™ 3080 V2 OC Edition 10GB GPU card (https://www.asus.com/fi/motherboards-components/graphics-cards/tuf-gaming/tuf-rtx3080-o10g-v2-gaming/) Is it because I have an GeForce card? (I hope not) 
Any suggestion on how to solve this?

thanks,

Andreas

Environment
```
anaconda (latest)
Python 3.10.9 (main, Mar  1 2023, 18:23:06) [GCC 11.2.0]
Jupyter notebook 6.5.2
IPython 8.8.0 -- An enhanced Interactive Python. Type '?' for help.
```

here's the full dump

```
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[2], line 5
      3 from bertopic_mod import BERTopicMod
      4 #if use_GPU:
----> 5 from cuml.cluster import HDBSCAN
      6 from cuml.manifold import UMAP

File ~/.local/lib/python3.10/site-packages/cuml/__init__.py:17
      1 #
      2 # Copyright (c) 2022, NVIDIA CORPORATION.
      3 #
   (...)
     14 # limitations under the License.
     15 #
---> 17 from cuml.internals.base import Base, UniversalBase
     19 # GPU only packages
     21 import cuml.common.cuda as cuda

File ~/.local/lib/python3.10/site-packages/cuml/internals/__init__.py:17
      1 #
      2 # Copyright (c) 2019-2022, NVIDIA CORPORATION.
      3 #
   (...)
     14 # limitations under the License.
     15 #
---> 17 from cuml.internals.base_helpers import (
     18     BaseMetaClass,
     19     _tags_class_and_instance
     20 )
     21 from cuml.internals.api_decorators import (
     22     _deprecate_pos_args,
     23     api_base_fit_transform,
   (...)
     35     exit_internal_api,
     36 )
     37 from cuml.internals.api_context_managers import (
     38     in_internal_api,
     39     set_api_output_dtype,
     40     set_api_output_type,
     41 )

File ~/.local/lib/python3.10/site-packages/cuml/internals/base_helpers.py:20
     17 from inspect import Parameter, signature
     18 import typing
---> 20 from cuml.internals.api_decorators import (
     21     api_base_return_generic,
     22     api_base_return_array,
     23     api_base_return_sparse_array,
     24     api_base_return_any,
     25     api_return_any,
     26     _deprecate_pos_args
     27 )
     28 from cuml.internals.array import CumlArray
     29 from cuml.internals.array_sparse import SparseCumlArray

File ~/.local/lib/python3.10/site-packages/cuml/internals/api_decorators.py:24
     21 import warnings
     23 # TODO: Try to resolve circular import that makes this necessary:
---> 24 from cuml.internals import input_utils as iu
     25 from cuml.internals.api_context_managers import BaseReturnAnyCM
     26 from cuml.internals.api_context_managers import BaseReturnArrayCM

File ~/.local/lib/python3.10/site-packages/cuml/internals/input_utils.py:19
      1 #
      2 # Copyright (c) 2019-2022, NVIDIA CORPORATION.
      3 #
   (...)
     14 # limitations under the License.
     15 #
     17 from collections import namedtuple
---> 19 from cuml.internals.array import CumlArray
     20 from cuml.internals.array_sparse import SparseCumlArray
     21 from cuml.internals.global_settings import GlobalSettings

File ~/.local/lib/python3.10/site-packages/cuml/internals/array.py:22
     19 import operator
     20 import pickle
---> 22 from cuml.internals.global_settings import GlobalSettings
     23 from cuml.internals.logger import debug
     24 from cuml.internals.mem_type import MemoryType, MemoryTypeError

File ~/.local/lib/python3.10/site-packages/cuml/internals/global_settings.py:20
     18 import threading
     19 from cuml.internals.available_devices import is_cuda_available
---> 20 from cuml.internals.device_type import DeviceType
     21 from cuml.internals.logger import warn
     22 from cuml.internals.mem_type import MemoryType

File ~/.local/lib/python3.10/site-packages/cuml/internals/device_type.py:19
      1 #
      2 # Copyright (c) 2022, NVIDIA CORPORATION.
      3 #
   (...)
     14 # limitations under the License.
     15 #
     18 from enum import Enum, auto
---> 19 from cuml.internals.mem_type import MemoryType
     22 class DeviceTypeError(Exception):
     23     '''An exception thrown to indicate bad device type selection'''

File ~/.local/lib/python3.10/site-packages/cuml/internals/mem_type.py:25
     19 from cuml.internals.device_support import GPU_ENABLED
     20 from cuml.internals.safe_imports import (
     21     cpu_only_import,
     22     gpu_only_import
     23 )
---> 25 cudf = gpu_only_import('cudf')
     26 cp = gpu_only_import('cupy')
     27 cpx_sparse = gpu_only_import('cupyx.scipy.sparse')

File ~/.local/lib/python3.10/site-packages/cuml/internals/safe_imports.py:366, in gpu_only_import(module, alt)
    340 '''A function used to import modules required only in GPU installs
    341 
    342 This function will attempt to import a module with the given name, but it
   (...)
    363     UnavailableMeta.
    364 '''
    365 if GPU_ENABLED:
--> 366     return importlib.import_module(module)
    367 else:
    368     return safe_import(
    369         module,
    370         msg=f'{module} is not installed in non GPU-enabled installations',
    371         alt=alt
    372     )

File ~/anaconda3/lib/python3.10/importlib/__init__.py:126, in import_module(name, package)
    124             break
    125         level += 1
--> 126 return _bootstrap._gcd_import(name[level:], package, level)

ModuleNotFoundError: No module named 'cudf'
---
```

**Comment by MaartenGr** [comment]
How did you install cuML? Also, I believe you can find several options for installing cuML/Rapids [here](https://docs.rapids.ai/install#selector) that should make it a bit easier.

**Comment by aph61** [comment]
Hi Maarten,

I did as you recommended. I installed the full anaconda suite (pip seems to give problems with cuML). After that I installed the software as instructed; installation generated a separate venv called (rapids-23.02), in which I launched jupyter notebook.
Unfortunately it didn't help. I closed the jupyter kernel, and started a new one in the base environment. Same result.
The only thing that might be different is that I work in a subdirectory of ~/Projects/, not in a subdir of anaconda3.

In both cases the import error was as before. 

hope this helps in giving a pointer

best,

Andreas
